### PR TITLE
enhance CI testing

### PR DESCRIPTION
**Pull request by bitsf**
add ci cases
1. install harbor-operator
2. install harbor
3. curl api
4. docker push

microk8s 1.17 doesn't support registry with ipv6

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -90,30 +90,63 @@ jobs:
...
+        set -ex
+        # make helm-install
+        sed -i '1inamespace: harbor-op' config/default/kustomization.yaml
+        kubectl create ns harbor-op
+        kustomize build config/default | kubectl apply -f -
+        kubectl wait --for=condition=Ready pod --all --timeout 100s -n harbor-op
+
+    - name: install harbor
+      run: |
+        export GITHUB_USER=xxx
```
Comment (steven-zou) [inline .github/workflows/tests.yml:126]:
```suggestion
     
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -90,30 +90,63 @@ jobs:
...
-        export KUBECONFIG="$(realpath "$(mktemp kubeconfig.XXXXX)")"
-        export USE_EXISTING_CLUSTER="true"
+        make go-test
 
-        sudo microk8s.config > "${KUBECONFIG}"
+    - name: install harbor-operator
+      run: |
+        set -ex
+        # make helm-install
+        sed -i '1inamespace: harbor-op' config/default/kustomization.yaml
```
Comment (steven-zou) [inline .github/workflows/tests.yml:119]: If we add the default namespace to Kustomization file, we may not need to do `sed` anymore, right?

**Review by steven-zou — COMMENTED** [review]
commented

File: .github/workflows/tests.yml (hunk 2 above)
Comment (bitsf) [reply .github/workflows/tests.yml]: fix

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -60,60 +60,124 @@ jobs:
...
+        for i in $(seq 1 6);do
+          sleep 30
+          echo $i
+          kubectl get all
+        done
+        if ! time kubectl wait --for=condition=Ready pod --all --timeout 600s ;then
+          echo install harbor failed
+          kubectl get all
+
+          for n in $(kubectl get po |grep -v Running|grep -v NAME|awk '{print $1}');do
```
Comment (steven-zou) [inline .github/workflows/tests.yml:151]: @bitsf 

I don't think we need to describe each pod here. We can find the unready pods and print out the log  by calling kubectl logs <POD>

**Review by steven-zou — COMMENTED** [review]
commented

File: .github/workflows/tests.yml (hunk 3 above)
Comment (bitsf) [reply .github/workflows/tests.yml:151]: it has `grep -v Running`

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**